pkg/virt-controller/watch/migration.go

```diff
@@ -808,6 +802,21 @@ func (c *MigrationController) handleTargetPodHandoff(migration *virtv1.VirtualMa
 		vmiCopy.Status.MigrationState.MigrationConfiguration = clusterMigrationConfigs
 	}
 
+	if vmiCopy.Status.MigrationState.MigrationConfiguration.AllowPostCopy != nil &&
+		*vmiCopy.Status.MigrationState.MigrationConfiguration.AllowPostCopy {
+		isPrivileged, err := psa.IsNamespacePrivilegedWithStore(c.namespaceStore, c.clientset, vmiCopy.Namespace)
+		if err != nil {
+			return err
+		}
+
+		if !isPrivileged {
+			doNotAllowPostCopy := false
+			vmiCopy.Status.MigrationState.MigrationConfiguration.AllowPostCopy = &doNotAllowPostCopy
+			log.Log.Object(vmi).Warningf("PostCopy disabled for migration %s/%s as the namespace is not privileged", migration.Namespace, migration.Name)
```

Comment:
I remember discussing this offline with Antonio, and rejecting migrations will cause all kinds of problems, notably breaking some upgrades.
Also, since the user technically just "allowed" (not mandated) post-copy migrations, it seems fair to run some without it, especially if it avoids catastrophic failure.
However, as you said in another comment, we do need a better test than just "is the namespace privileged?"